manifests/custom_config.pp

```diff
@@ -115,9 +116,10 @@
       require     => Anchor['::apache::modules_set_up'],
     }
 
+    $remove_command = ['/bin/rm', $_file_path]
     exec { "remove ${name} if invalid":
-      command     => "/bin/rm ${confdir}/${_filename}",
-      unless      => $verify_command,
+      command     => $remove_command,
+      unless      => [$verify_command],
```

Comment:
Bit iffy on doing this.
The verify command can be manually set and we know that splitting the command can cause errors in certain situations, i.e. the use of pipes
Not sure I like the idea of splitting the command when I don't know what it may be.
Probably other thinking it, but still

Reply:
Yes that is a good point.. but to me bolsters the fact that we should be guiding people to properly structure their commands.

`unless` takes an array of arrays. Each inner array is interpreted as a command by the provider. When multiple commands are passed each one needs to evaluate as true for the overall success of `unless.

Though the above would be a breaking change for some which adds to the complication.

Reply:
Could split the default command into an array. Then set verify_command to be: `Variant[String, Array[String], Array[Array[String]]`
It's a bit messy, but is technically what it should accept

Reply:
Then maybe remove the string in the next major release

Reply:
Yeah I think that is a nice compromise actually given that the exec type defines `unless` like [this](https://github.com/puppetlabs/puppet/blob/04cf2ae3ac98d7ac628663f4ef0b652b6b013041/lib/puppet/type/exec.rb#L451).

Reply:
pushed

Reply:
> Could split the default command into an array. Then set verify_command to be: Variant[String, Array[String], Array[Array[String]]
> It's a bit messy, but is technically what it should accept

I like this, but would it make sense to (also) introduce a `Stdlib::Exec::Unless` data type? While we couldn't use it immediately, it does signal to the user how it should be used. Just like there's `Stdlib::Ensure::*`.

Reply:
also rebased onto main

Reply:
and squashed down all my corrective changes into one commit

Reply:
@ekohl that would be nice yes. I'm a fan of anything we can do to help users on their journey.